sdk/src/main/java/com/mapbox/maps/ResourceOptionsManager.kt

```diff
@@ -0,0 +1,42 @@
+package com.mapbox.maps
+
+import android.content.Context
+
+/**
+ * Convenience class that holds the default ResourceOptions.
+ * `ResourceOptionsManager` could be created per `MapView` instance if given view should use some specific resource options.
+ * Most common use-case when all `MapView`s should use same resource options could be handled by using `ResourceOptionsManager.getDefault(context: Context)` static object.
+ *
+ * @property resourceOptions the initial resource options.
+ */
+class ResourceOptionsManager(
+  var resourceOptions: ResourceOptions
```

Comment:
nit:
it can be replaced with 
```suggestion
  private val resourceOptionsBuilder: ResourceOptions.Builder
```
`update` will be simplified
```
fun update(block: ResourceOptions.Builder.() -> Unit) {    
       resourceOptionsBuilder.apply(block).build()
```
`fun resourceOptions() = resourceOptionsBuilder.build()`

Reply:
The idea here is that the resourceOptions will be readable and writable for customers, user could set the default ResourceOptions using 
```
ResourceOptionsManager.getDefault(context).resourceOptions = ResourceOptions.Builder().applyDefaultParams(context).build()
```